pkg/processor/k8sprocessor/kube/client.go

```diff
@@ -117,6 +119,51 @@ func New(
 	return c, err
 }
 
+// OwnerFeedbackLoop starts a loop that will receive notifications about updates
+// to endpoints objects which are then used to update corresponding pods with
+// ownership info.
+// For info it only updates the services attribute. See
+// Test_PodIsAnnotatedWithOwnershipInfoImmediatelyAfterOwnersAreCreated test in
+// client_test.go for details.
+func (c *WatchClient) OwnerFeedbackLoop(stopCh <-chan struct{}) {
+	ch := c.op.GetPodUpdateChan()
+	for {
+		select {
+		case <-stopCh:
+			return
+
+		case pod, ok := <-ch:
+			if !ok {
+				return
+			}
+
+			p, ok := c.GetPod(generatePodID(pod))
+			if !ok {
+				continue
+			}
+
+			if c.Rules.OwnerLookupEnabled {
+				if c.Rules.ServiceName {
+					services := c.op.GetServices(pod.Name)
+					if len(services) > 0 {
+						c.m.Lock()
+						// TODO: this cannot be done this way because Attributes are
+						// being accessed elsewhere e.g.
+						// * in kubernetesprocessor.getAttributesForPod()
+						// * in tests when assertions on attributes are being made.
+						//
+						// which create data race situations.
```

Comment:
Is this a problem because other contexts hold references to the Pod object? If so, can't we make a copy of it, replace the Attributes, and store the copy? That's what `addOrUpdatePod` does, in effect, although it creates a new object from the data it gets, as opposed to copying the existing one.

Reply:
Yes but as of now there's no problem with that because `Attributes` is only accessed in

* https://github.com/SumoLogic/sumologic-otel-collector/blob/c7750165c933ebbacd3079f0693ce4b2d7ab5951/pkg/processor/k8sprocessor/kube/client.go#L407
* and https://github.com/SumoLogic/sumologic-otel-collector/blob/2dcbc25d9907bdb219e03a18f6a584b62a1c9f32/pkg/processor/k8sprocessor/processor.go#L129

where in the former it's created and then saved in `c.Pod` under a lock and in the latter it's just retrieved to copy into the processed record.

Reply:
So it's not actually a problem? I'm not sure what the point of this TODO is, in that case. Why not fix it right now if it's as simple as copying the Pod?